lighthouse-core/audits/resource-summary.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+const NetworkRecords = require('../computed/network-records.js');
+const i18n = require('../lib/i18n/i18n.js');
+const MainResource = require('../computed/main-resource.js');
+const URL = require('../lib/url-shim');
+const Util = require('../report/html/renderer/util.js');
+
+const UIStrings = {
+  /** Imperative title of a Lighthouse audit that tells the user to minimize the size and quantity of resources used to load the page. */
+  title: 'Keep request counts and file sizes small',
+  /** Description of a Lighthouse audit that tells the user that they can setup a budgets for the quantity and size of page resources. No character length limits. */
+  description: 'To set budgets for the quantity and size of page resources,' +
+    ' add a budgets.json file.',
+  /** [ICU Syntax] Label for an audit identifying the number of requests and kilobytes used to load the page. */
+  displayValue: `{requestCount, plural, =1 {1 request} other {# requests}} • { formattedByteCount}`,
+  /** Label for the combination of all resources on the page. */
+  totalResourceType: 'Total',
+  /** Label for the 'Document' resource type. */
+  documentResourceType: 'Document',
+  /** Label for the 'script' resource type. */
+  scriptResourceType: 'Script',
+  /** Label for the 'stylesheet' resource type. */
+  stylesheetResourceType: 'Stylesheet',
+  /** Label for the 'image' resource type. */
+  imageResourceType: 'Image',
+  /** Label for the 'media' resource type. */
+  mediaResourceType: 'Media',
+  /** Label for the 'font' resource type. */
+  fontResourceTyoe: 'Font',
+  /** Label for the 'other' resource type. */
+  otherResourceType: 'Other',
+  /** Label for the 'third-party' resource type. */
+  thirdPartyResourceType: 'Third-party',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class ResourceSummary extends Audit {
+  /**
+       * @return {LH.Audit.Meta}
+       */
+  static get meta() {
+    return {
+      id: 'resource-summary',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      requiredArtifacts: ['devtoolsLogs'],
+    };
+  }
+
+  /**
+     * @param {LH.Artifacts.NetworkRequest} record
+     * @return string
```

Comment:
could we get a `export type BudgetsResourceType = 'image'|'stylesheet'|etc` going? :D

Reply:
Good idea. I think think `budgets.d.ts` is probably the best place for that to go so I will add that to this PR once https://github.com/GoogleChrome/lighthouse/pull/8427  is merged.